### PR TITLE
JAMES-2143 Download messages

### PR DESCRIPTION
**Pull request by chibenwa**
Left to do:
 - Auto review more carefully
 - Check there is nothing more to do
 - Pass CI tests

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/Blob.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+public class Blob {
+
+    public static class Builder {
+        private BlobId blobId;
+        private byte[] payload;
+        private String contentType;
+
+        private Builder() {
+            blobId = null;
```
Comment (aduprat) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/Blob.java]: Not needed

File: server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java [hunk 2]
```diff
@@ -130,7 +145,11 @@ public void optionDownload(String username) throws Throwable {
 
     @When("^\"([^\"]*)\" downloads \"([^\"]*)\"$")
     public void downloads(String username, String attachmentId) throws Throwable {
```
Comment (aduprat) [inline server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/DownloadStepdefs.java]: s/attachmentId/blobId

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreBlobManager.java [hunk 3]
```diff
@@ -57,6 +57,11 @@ public StoreBlobManager(AttachmentManager attachmentManager, MessageIdManager me
     }
 
     @Override
+    public BlobId toBlobId(final MessageId messageId) {
```
Comment (aduprat) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreBlobManager.java]: You don't need `final`

File: mailbox/api/src/test/java/org/apache/james/mailbox/model/BlobTest.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        assertThatThrownBy(() ->
+            Blob.builder()
+                .contentType(CONTENT_TYPE)
+                .payload(PAYLOAD)
+                .build())
+            .isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void buildShouldThrowOnContentType() {
```
Comment (rouazana) [inline mailbox/api/src/test/java/org/apache/james/mailbox/model/BlobTest.java]: *OnMissing

File: mailbox/api/src/test/java/org/apache/james/mailbox/model/BlobTest.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        assertThatThrownBy(() ->
+            Blob.builder()
+                .id(ID)
+                .payload(PAYLOAD)
+                .build())
+            .isInstanceOf(IllegalStateException.class);
+    }
+
+    @Test
+    public void buildShouldThrowOnPayload() {
```
Comment (rouazana) [inline mailbox/api/src/test/java/org/apache/james/mailbox/model/BlobTest.java]: *OnMissing

**Comment by chibenwa** [comment]
:+1:   thanks @aduprat !

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
Merge: https://github.com/linagora/james-project/pull/975

**Pull request closed without merging**